aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -780,6 +782,121 @@ static void linalg_lu_factor_out_mps_impl(const Tensor& A, bool pivot, Tensor& L
   return out;
 }
 
+static Tensor& linalg_cholesky_mps_impl(const Tensor& input, bool upper, Tensor& out) {
+  using namespace mps;
+
+  TORCH_CHECK(out.is_mps());
+  TORCH_CHECK(input.scalar_type() == at::ScalarType::Float, "linalg.cholesky: Input tensor must be float32");
+  TORCH_CHECK(input.dim() >= 2, "linalg.cholesky: Input tensor must be at least 2D");
+  TORCH_CHECK(input.size(-2) == input.size(-1), "linalg.cholesky: Input tensor must be square");
+
+  if (input.numel() == 0 || out.numel() == 0) {
+    out.zero_();
+    return out;
+  }
+
+  int64_t ndim = out.dim();
+  int64_t N = out.size(-1);
+  int64_t B = 1;
+  for (int64_t i = 0; i < ndim - 2; i++) {
+    B *= out.size(i);
+  }
+
+  uint32_t batch_stride = N * N;
+
+  auto stream = getCurrentMPSStream();
+  auto device = MPSDevice::getInstance()->device();
+
+  auto factorDiagonalPSO = lib.getPipelineStateForFunc("factorDiagonalBlock");
+  auto applyTRSMPSO = lib.getPipelineStateForFunc("applyTRSM");
+  auto applySYRKPSO = lib.getPipelineStateForFunc("applySYRK");
+
+  int64_t NB = std::min<int64_t>(32, N);
+  int64_t numBlocks = (N + NB - 1) / NB;
+
+  Tensor success = at::empty({B}, input.options().dtype(kInt)).fill_(1);
+  id<MTLBuffer> successBuffer = getMTLBufferStorage(success);
+
+  MTLSize threadGroupSize = MTLSizeMake(256, 1, 1);
+  id<MTLBuffer> outBuffer = getMTLBufferStorage(out);
+  id<MTLComputeCommandEncoder> computeEncoder = stream->commandEncoder();
+  [computeEncoder setBuffer:outBuffer offset:0 atIndex:0];
+
+  @autoreleasepool {
+    dispatch_sync_with_rethrow(stream->queue(), ^() {
+      for (int64_t k = 0; k < numBlocks; k++) {
+        uint32_t activeNB_k = (uint32_t)std::min(N - k * NB, (int64_t)NB);
+        [computeEncoder setComputePipelineState:factorDiagonalPSO];
+        std::array<uint32_t, 6> sizes = {static_cast<uint32_t>(N),
+                                         static_cast<uint32_t>(NB),
+                                         static_cast<uint32_t>(k),
+                                         static_cast<uint32_t>(activeNB_k),
+                                         static_cast<uint32_t>(B),
+                                         static_cast<uint32_t>(batch_stride)};
+        id<MTLBuffer> sizesBuffer = [device newBufferWithBytes:sizes.data()
```

Comment:
This will cause a memory leak, wouldn't it? (I would prefer to just allocate temporary tensor rather than use autorelease pool here)